library/pkparse.c

```diff
@@ -1343,7 +1361,9 @@ int mbedtls_pk_parse_key( mbedtls_pk_context *pk,
      * also ok and in line with the mbedtls_pk_free() calls
      * on failed PEM parsing attempts. */
 
-    return( MBEDTLS_ERR_PK_KEY_INVALID_FORMAT );
+    if( ret & MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED )
```

Comment:
Masking an error code doesn't make sense. Did you mean “if the low-level error is feature-unsupported then mask out the high-level error”? That would be
```
    if( ( ret & -0x7f ) == MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED )
```